gpii/node_modules/preferencesServer/src/preferencesService.js

```diff
@@ -73,6 +125,45 @@ var fluid = fluid || require("infusion");
                 listener: "gpii.preferencesServer.preferencesService.updateGpiiKey",
                 args: ["{dataStore}.updateGpiiKey", "{arguments}.0"],
                 namespace: "updateGpiiKey"
+            }],
+            onCloudSafeCredCreate: [{
+                func: "{dataStore}.findById",
+                args: ["{arguments}.0"],
+                namespace: "findById"
+            }, {
+                func: "gpii.dbOperation.verifyExists",
+                args: ["{arguments}.0", "prefsSafe", "The prefs safe does not exist"]
+            }, {
+                func: "{expressUserUtils}.createNewUser",
```

Comment:
What if the prefs safe already has an existing username? Shall this func [check the existing user account](https://github.com/GPII/gpii-express-user/blob/master/src/js/server/signup.js#L21) before creating a new one?

Reply:
I think we should hold off for now on adding any type of limitation like that.  I think part of the beauty of our credential/key document structure is that you can have multiple keys/credentials of various types, and choose to revoke them, delete them, have multiple legit ones at the same time.  As of now, this setup is compatible with how gpii-express-user works, and we could have more than 1 legitimate username/password login for a prefssafe if a use case for that presented itself.

I imagine as the "Morphic World" UI design moves along, and someone works through the raminifications of deployments we might have more guidance on this, but I'd like to leave it flexible for now.

Reply:
Thanks for the explanation. It makes sense.